src/utils.ts

```diff
@@ -18,17 +28,45 @@ export const getSubtypeAndSridFromModifier = (
   if (!isGeography && srid !== 0) {
     throw new Error("Unexpected SRID with geometry type");
   }
-  const subtype = (modifier & 255) >> 2;
-  if (subtype > 7 || subtype < 0) {
+  if (
+    subtype !== 0 &&
+    subtype !== 1 &&
+    subtype !== 2 &&
+    subtype !== 3 &&
+    subtype !== 4 &&
+    subtype !== 5 &&
+    subtype !== 6 &&
+    subtype !== 7
```

Comment:
Yup

Reply:
This could possibly read better as a switch / case (with default being the error), or with an `if (parseInt(subtype,10) !== subtype || subtype <0 || subtype >7)` - sadly supporting Javascript developers makes this more complicated :-P